File: golang/1_two-sum.go
// hash 法， 4 ms, 4.02 MB
func twoSum(nums []int, target int) []int {
	temp_map = map[int]int{}
	for index1, value1 := range nums {
		if index2, ok := temp_map[target-value1]; ok {
			return []int{index1, index2}
		}
		temp_map[value1] = index1
	}
	return nil
}

File: python3/1_two-sum.py
# 暴力法
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for index1, value1 in enumerate(nums):
            for index2, value2 in enumerate(nums):
                if index2 <= index1:
                    continue
                temp_sum = value1 + value2
                if temp_sum == target:
                    return [index1, index2]
        return []


# hash 法, 44 ms, 18.15 MB
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        temp_dict = {}
        for index, value in enumerate(nums):
            if target - value in temp_dict:
                return [index, temp_dict.get(target - value)]
            temp_dict[value] = index
        return []